### DISCUSSION TITLE
Locking among different branches for the same working copy

### DISCUSSION THREAD

**Discussion by SalmaFarghaly** [git-lfs/git-lfs#5004] [Q&A]
Hello,
Does git lfs support locking among different branches in the same working copy? 
I have tried to do the following to check that:
1) create file "file.txt" on main branch 
2) git add "file.txt"
3) git lfs track "file.txt"
4) git lfs track "file.txt" --lockable
5) git add .gitattributes
6) git commit --m "commit"
7) git lfs lock "file.txt", Until now I have managed to take the lock on "file.txt" and the file becomes writable in my workspace
8) git checkout -b new
Now I can see that "file.txt" is editable in "new" branch. I expect that "file.txt" will be read-only on "new" branch as I have taken the lock on the file in "main" branch.

**Comment by bk2204** [comment]
Hey,

Locks are not specific to a particular branch.  The idea is that you lock the file to prevent others from modifying it, and then you can merge that change into the main branch without others having made similar modifications.  If we restricted locks to a particular branch, someone else could modify that file on their own branch, and then cause a conflict without notice.

**Comment by SalmaFarghaly** [comment]
Hey @bk2204 Thank you for your response.
I agree on that locks should not be restricted to a particular branch. I was asking about the case that if user A takes a lock on File F in branch B. I can see that user A is still able to modify File F on another branch C. I expect when user A takes a lock on File F in branch B . he won't be able to edit the file on any other branch except B.

**Reply by bk2204** [reply]
No, it's not the case that the user should be able to modify the file only on a specific branch.  It may happen that the user decides that they need to move their changes to another branch or to rebase one branch onto another, and we wouldn't want to prevent them from doing that.
